File: StartAppFrontend/app/src/main/java/com/skyletto/startappfrontend/data/database/dao/TagDao.java
package com.skyletto.startappfrontend.data.database.dao;

import com.skyletto.startappfrontend.domain.entities.Tag;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

@Dao
public interface TagDao{
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long add(Tag tag);

    @Query("select * from tags")
    List<Tag> getAll();

    @Query("select * from tags where id=:id")
    Tag getById(Long id);
}

File: StartAppFrontend/app/src/main/java/com/skyletto/startappfrontend/data/database/dao/TagDao.kt
package com.skyletto.startappfrontend.data.database.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import com.skyletto.startappfrontend.domain.entities.Tag
import io.reactivex.Single

@Dao
interface TagDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun add(tag: Tag): Single<Long>

    @Query("select * from tags")
    fun getAll(): LiveData<List<Tag>>

    @Query("select * from tags where id=:id")
    fun getById(id: Long): Single<Tag>

    @Query("delete from tags")
    fun removeAll():Single<Int>
}